src/helpers/elements-interactions.ts

```diff
@@ -9,6 +10,24 @@ async function waitUntilElementDisappear(page: Page, elementSelector: string, ti
   await page.waitForSelector(elementSelector, { hidden: true, timeout });
 }
 
+// TODO: not sure if it should be here or in watiting.ts
+// TODO: Can be used also in visa-cal
+async function waitUntilIframeFound(page: Page, framePredicate: (frame: Frame) => boolean, timeout = 30000) {
+  let frame: Frame | undefined;
+  await waitUntil(() => {
+    frame = page
+      .frames()
+      .find(framePredicate);
+    return Promise.resolve(!!frame);
+  }, 'wait for iframe with login form', timeout, 1000);
+
+  if (!frame) {
+    throw new Error('failed to find iframe');
+  }
+
+  return frame;
+}
```

Comment:
- I don't know if it should be implemented here or in `waiting.ts`.
- I think it is convenient to combine the wait and the getElement.
- The implementation is copied from `visa-cal` and can be used there too.

Reply:
we can keep it here for now imo.